### PR TITLE
feat: add autoware_api.launch.xml to launch files in autoware_launch package

### PR DESCRIPTION
**Pull request by mitsudome-r**
Signed-off-by: mitsudome-r <ryohsuke.mitsudome@tier4.jp>

## Description
This PR modifies the following launch files to launch api related packages to have integration with TIER IV's scenario simulator.
* autoware.launch.xml
* logging_simulator.launch.xml
* planning_simulator.launch.xml

## Related links
* Issue: https://github.com/autowarefoundation/autoware.universe/issues/656
Dependent PR:
* https://github.com/autowarefoundation/autoware/pull/156
* https://github.com/autowarefoundation/autoware.universe/pull/658

## Tests performed
Here's the test procedure: 

### Setup
* Download and unzip the [sample data](https://drive.google.com/file/d/1vWMLbmwJJE5tYO40ypCMxqtmgQPQxhiw/view?usp=sharing) that includes the map and a sample scenario file
* modify the map file path in the scenario file (`t4v2.yaml`) to the path of the Downloaded map
* If dependent PRs are not merged, checkout the feature branch
  * `cd autoware`
  * Checkout `mitsudome-r:feature/add-ad-api-adaptor` branch in autoware ([reference](https://github.com/autowarefoundation/autoware/pull/156))
  * `vcs pull src < autoware.proj.repos`
  * `cd src/universe/autoware`
  *  Checkout `mitsudome-r:feature/add-tier4-autoware-api-launch` branch in autoware ([reference](https://github.com/autowarefoundation/autoware.universe/pull/658))
* Also pull scenario simulator
  * `cd ~/autoware`
  * `vcs import src < simulator.repos`
* Build Autoware 
  * `cd ~/autoware`
  * `colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release`

### Testing Scenario Simulation
Terminal1: (modify the map path to the downloaded map)
```
source ~/autoware/install/setup.bash
ros2 launch autoware_launch planning_simulator.launch.xml map_path:=$HOME/Downloads/sample_data/map/ vehicle_model:=sample_vehicle sensor_model:=sample_sensor_kit scenario_simulation:=true
```

Terminal2: 
```
source ~/autoware/install/setup.bash
ros2 launch scenario_test_runner scenario_test_runner.launch.py sensor_model:=sample_sensor_kit vehicle_model:=sample_vehicle scenario:=/home/mitsudome-r/Downloads/sample_data/t4v2.yaml architecture_type:=awf/universe launch_rviz:=true launch_autoware:=false
```

You should see the vehicle drive in straight line after waiting a while (about 30 sec).
Screenshot from 2022-04-08 19-30-59

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: autoware_launch/launch/autoware.launch.xml [hunk 1]
```diff
@@ -92,6 +92,12 @@
     </include>
   </group>
 
+  <!-- API -->
+  <include file="$(find-pkg-share tier4_autoware_api_launch)/launch/autoware_api.launch.xml">
+    <arg name="init_simulator_pose" value="true"/>
+    <arg name="init_localization_pose" value="false"/>
+  </include>
+
```
Comment (kenji-miyake) [inline autoware_launch/launch/autoware.launch.xml:100]: https://github.com/tier4/autoware_launch/blob/d629633e3c9533fb54e715eca29a6c0aded56661/autoware_launch/launch/autoware.launch.xml#L92-L98

File: autoware_launch/launch/planning_simulator.launch.xml [hunk 2]
```diff
@@ -84,6 +84,12 @@
     <include file="$(find-pkg-share web_controller)/launch/web_controller.launch.xml" unless="$(var scenario_simulation)"/>
   </group>
 
+  <!-- API -->
+  <include file="$(find-pkg-share tier4_autoware_api_launch)/launch/autoware_api.launch.xml">
+    <arg name="init_simulator_pose" value="true"/>
+    <arg name="init_localization_pose" value="false"/>
+  </include>
+
```
Comment (kenji-miyake) [inline autoware_launch/launch/planning_simulator.launch.xml:92]: https://github.com/tier4/autoware_launch/blob/d629633e3c9533fb54e715eca29a6c0aded56661/autoware_launch/launch/planning_simulator.launch.xml#L69-L75

File: autoware_launch/launch/logging_simulator.launch.xml [hunk 3]
```diff
@@ -107,6 +107,12 @@
     </include>
   </group>
 
+  <!-- API -->
+  <include file="$(find-pkg-share tier4_autoware_api_launch)/launch/autoware_api.launch.xml">
+    <arg name="init_simulator_pose" value="true"/>
+    <arg name="init_localization_pose" value="false"/>
+  </include>
+
```
Comment (kenji-miyake) [inline autoware_launch/launch/logging_simulator.launch.xml:115]: https://github.com/tier4/autoware_launch/blob/d629633e3c9533fb54e715eca29a6c0aded56661/autoware_launch/launch/logging_simulator.launch.xml#L108-L114

**Review by kenji-miyake — COMMENTED** [review]
Added links to the copy sources.

File: autoware_launch/launch/autoware.launch.xml (hunk 1 above)
Comment (kenji-miyake) [reply autoware_launch/launch/autoware.launch.xml:100]: Flipped?
image

File: autoware_launch/launch/logging_simulator.launch.xml (hunk 3 above)
Comment (kenji-miyake) [reply autoware_launch/launch/logging_simulator.launch.xml:115]: Flipped?
image

**Comment by kenji-miyake** [comment]
@isamu-takagi Could you confirm the flags?

**Comment by isamu-takagi** [comment]
@kenji-miyake  Yes, the items pointed out are right.

When using localization but not simulator (autoware, logging_simulator)
```
<arg name="init_simulator_pose" value="false"/>
<arg name="init_localization_pose" value="true"/>
```
When the simulator manage the vehicle pose (planning_simulator)
```
<arg name="init_simulator_pose" value="true"/>
<arg name="init_localization_pose" value="false"/>
```

If using simulator and localization such as end-to-end simulation, both will be true.

**Comment by angry-crab** [comment]
`vcs pull src < autoware.proj.repos`
Should this be `vcs import src < autoware.repos` instead?

**Comment by angry-crab** [comment]
Getting this when building `traffic_simulator`
`/usr/bin/ld: ../../../libtraffic_simulator.so: undefined reference to `quaternion_operation::convertQuaternionToEulerAngle(geometry_msgs::msg::Quaternion_<std::allocator<void> >)'
`
`quatuernion_operation` has been installed by rosdep install.

File: autoware_launch/launch/autoware.launch.xml (hunk 1 above)
Comment (mitsudome-r) [reply autoware_launch/launch/autoware.launch.xml:100]: Thank you, I have fixed the arguments.

File: autoware_launch/launch/planning_simulator.launch.xml (hunk 2 above)
Comment (mitsudome-r) [reply autoware_launch/launch/planning_simulator.launch.xml:92]: For planning simulator, it seems like the booleans are matching with the original code. (and it seems to match with what @isamu-takagi said in https://github.com/autowarefoundation/autoware_launch/pull/26#issuecomment-1093877404)

Comment (kenji-miyake) [reply autoware_launch/launch/planning_simulator.launch.xml:92]: Yes, I'll resolve this comment.

**Comment by kenji-miyake** [comment]
:thinking: 

```
❯ ros2 launch scenario_test_runner scenario_test_runner.launch.py sensor_model:=sample_sensor_kit vehicle_model:=sample_vehicle scenario:=$HOME/Downloads/sample_data/t4v2.yaml architecture_type:=awf/universe launch_rviz:=true launch_autoware:=false
[INFO] [launch]: All log files can be found below /home/kenji/.ros/log/2022-04-12-20-55-54-575445-desktop-78281
[INFO] [launch]: Default logging verbosity is set to INFO
architecture_type       := awf/universe
autoware_launch_file    := planning_simulator.launch.xml
autoware_launch_package := autoware_launch
global_frame_rate       := 30.0
global_real_time_factor := 1.0
global_timeout          := 180
initialize_duration     := 30
launch_autoware         := false
launch_rviz             := true
output_directory        := /tmp
port                    := 8080
record                  := True
scenario                := /home/kenji/Downloads/sample_data/t4v2.yaml
sensor_model            := sample_sensor_kit
vehicle_model           := sample_vehicle
workflow                := /dev/null
sigterm_timeout         := 8
[INFO] [scenario_test_runner-1]: process started with pid [78307]
[INFO] [simple_sensor_simulator_node-2]: process started with pid [78309]
[INFO] [openscenario_interpreter_node-3]: process started with pid [78311]
[INFO] [openscenario_visualization_node-4]: process started with pid [78313]
[INFO] [rviz2-5]: process started with pid [78315]
[scenario_test_runner-1] [INFO 1649764556.012403148] [simulation.scenario_test_runner]: Run t4v2.xosc (1 of 1)
[openscenario_interpreter_node-3] [WARN 1649764558.019655055] [rcl.logging_rosout]: Publisher already registered for provided node name. If this is due to multiple nodes with the same name then all logs for that logger name will go out over the existing publisher. As soon as any node with that name is destructed it will unregister the publisher, preventing any further logs for that name from being published on the rosout topic.
[openscenario_interpreter_node-3] [WARN 1649764558.245283210] []: Error occurred while doing error handling.
[openscenario_interpreter_node-3] [WARN 1649764558.245316241] [rcl_lifecycle]: No transition matching 4 found for current state errorprocessing
[openscenario_interpreter_node-3] [ERROR 1649764558.245327891] []: Unable to start transition 4 from current state errorprocessing: Transition is not registered., at /tmp/binarydeb/ros-galactic-rcl-lifecycle-3.1.2/src/rcl_lifecycle.c:355
[scenario_test_runner-1] [ERROR 1649764558.247057913] [simulation.scenario_test_runner]: Interpreter is unconfigured now, but inactive expected.
[scenario_test_runner-1] [INFO 1649764558.247474189] [simulation.scenario_test_runner]: Shutdown interpreter.
[scenario_test_runner-1] /home/kenji/Downloads/sample_data/t4v2.yaml
[INFO] [scenario_test_runner-1]: process has finished cleanly [pid 78307]
```

**Comment by yamacir-kit** [comment]
@mitsudome-r @kenji-miyake 
I have just confirmed that scenario_simulator_v2:master now works with autowarefoundation/autoware:main by applying this pull request.

The sample scenario included in scenario_simulator_v2 was used for testing. The commands used are as follows

```
ros2 launch scenario_test_runner scenario_test_runner.launch.py \
   architecture_type:=awf/universe \
   record:=false \
   scenario:='$(find-pkg-share scenario_test_runner)/scenario/sample.yaml' \
   sensor_model:=sample_sensor_kit \
   vehicle_model:=sample_vehicle
```

(The map for the sample scenario is also included in scenario_simulator_v2)

**Pull request merged**